### ISSUE TITLE
Waiting on force close, attempt to force close again brings error: rpc error: code = Unknown desc = cannot close channel with state: ChanStatusLocalDataLoss|ChanStatusRestore

### ISSUE THREAD

**Issue by JustAGuy12345** [lightningnetwork/lnd#5693]
### Background

Trying to force close a channel using lnd.

### Your environment

latest stable version LND on Umbrel OS lightning node
raspberry pi 8gb

### Steps to reproduce

This channel originally would not loop out or complete a cooperative close. There was an OS level error following this and I had to recover my channels through backup. The channel backup initiated a force close on this channel and it has remained in waiting ever since.

### Expected behaviour

I am expecting a force close and deposit of the 600,000 satoshi balance into my bitcoin wallet from the channel close.

### Actual behaviour

lnd --forceclose emits an error and pending force close is not fulfilled.

**Comment by guggero** [comment]
You cannot force-close a channel that you recovered from the channel backup. You need to connect to the remote node and wait for it to force-close it.

**Comment by JustAGuy12345** [comment]
if that is true why did all 5 other channels have successful force closes and this one did not? Also what do you mean connect to the remote node? In which way should I connect to it? Thanks for closing this issue even though it is 100% still not solved on my end, super disrespectful.

**Comment by guggero** [comment]
> Also what do you mean connect to the remote node?

Look up the node public key in the `lncli pendingchannels` output, then go to a public network explorer (for example 1ml.com) and find out the IP/Tor address of the node. Then use `lncli connect <pubkey>@IP:port` to connect to them.

> Thanks for closing this issue

Yeah, that was probably an accident.

**Comment by JustAGuy12345** [comment]
it says I am already connected to this peer.

**Comment by guggero** [comment]
Okay, that's weird. Can you please send the latest log file of `lnd`? For some reason the remote node isn't force-closing the channel. Is it a well-known node on the other end? This isn't the first time I'm hearing that a node isn't assisting with the recovery even though there is a connection...

**Comment by guggero** [comment]
You can DM me the logs on Slack if you want, I have the same username there.

**Comment by JustAGuy12345** [comment]
The node is coincharge, they seem to be a fairly legitimate service. I messaged their email asking them to close on their end and they sort of refused in a confused manner. Sending logs next

**Comment by JustAGuy12345** [comment]
Sorry Im a very new user, what command should I do to get a log that I can paste for you?

**Comment by JustAGuy12345** [comment]
Nevermind Ill just extract it and send it through slack

**Comment by JustAGuy12345** [comment]
what is your email to send through slack?

**Comment by guggero** [comment]
Not sure what you mean. I'm `guggero` on this Slack instance: https://lightning.engineering/slack.html

**Comment by guggero** [comment]
Update: I took a look at the log and the DLP is requested from the remote node as expected. But it seems the remote node is not responding, the same way it seems to be described in https://github.com/lightningnetwork/lnd/issues/5604. I'm starting to believe this might be a bug in `lnd`. But without logs from the other side, it's very hard to tell.

I'm going to keep this issue open in the hope that we can shed some light on this but am going to change the title.

**Comment by guggero** [comment]
From the investigation done in https://github.com/lightningnetwork/lnd/issues/5604 it looks like `lnd` isn't responding to DLP if the channel is in the `waiting_close` state, from a failed coop-closure attempt (which is probably related to https://github.com/lightningnetwork/lnd/issues/5385 and https://github.com/lightningnetwork/lnd/issues/4819).

I'm going to see if we can simply follow DLP even if a channel is already in the `waiting_close` state.

**Comment by JustAGuy12345** [comment]
If so please give me the instructions on what I should do, thanks

**Comment by Crypt-iQ** [comment]
@guggero I don't think it's related to #5385 - that's a separate issue with coop close due to lack-of-locking which is fixed in https://github.com/lightningnetwork/lnd/pull/5618.  If the channel has been marked closed in the database (e.g. from DLP), we won't load a link into the htlcswitch.  It is expected that the remote force closes at this point.  If there's anything I can do to help let me know.  We don't load the link into the htlcswitch for safety reasons at this point

**Comment by guggero** [comment]
> I don't think it's related to #5385

I think the reason the channel was in `waiting_close` is that a coop-close was attempted but didn't go through. Either because one of the nodes was slow to respond or because of the locking issue. But _how_ the channel got into that state isn't the main issue here.
The fact that `lnd` doesn't respond to DLP when the channel is waiting to be closed is not optimal. I think if we discover remote DLP being requested, we should also look at the channel DB and not the htlcswitch only.
If you want to pick this up, feel free to do so! Otherwise I'm going to take a look later this week.

**Comment by Crypt-iQ** [comment]
> > I don't think it's related to #5385
> 
> I think the reason the channel was in `waiting_close` is that a coop-close was attempted but didn't go through. Either because one of the nodes was slow to respond or because of the locking issue. But _how_ the channel got into that state isn't the main issue here.
> The fact that `lnd` doesn't respond to DLP when the channel is waiting to be closed is not optimal. I think if we discover remote DLP being requested, we should also look at the channel DB and not the htlcswitch only.
> If you want to pick this up, feel free to do so! Otherwise I'm going to take a look later this week.

I think the DLP logic could be extracted into its own state-machine which is consulted instead of htlcswitch.  Maybe this is a longer-term change, and an easier sol'n is possible in short term.

**Comment by otaviobps** [comment]
Hey guys I had the same problem with more channels. I was migrating my debian setup to my raspiblitz. I started closing all channels, and force closing the channels I had with inactive peers. Before the 144 blocks, I recovered my backup in my raspiblitz. It was supposed to work just fine, with all channels closed and my funds in my wallet. However, some channels got stuck in a `waiting_close_channels` for ever. One of them is with Bitrefill. I asked them to force close it and send me the logs, but there are more channels that I'm trying to figure it out who are the owners:

```
{
    "total_limbo_balance": "0",
    "pending_open_channels": [
    ],
    "pending_closing_channels": [
    ],
    "pending_force_closing_channels": [
    ],
    "waiting_close_channels": [
        {
            "channel": {
                "remote_node_pub": "02c29b89b2121b2c1fa2e5422bc70e0bb7ae7326c7a9d2b796ed6b89cdc5a2871b",
                "channel_point": "cf00515a2d247838c9180e1335530a8b936d4eb9bbe14841126ff35f08f0c271:1",
                "capacity": "3759252",
                "local_balance": "0",
                "remote_balance": "0",
                "local_chan_reserve_sat": "0",
                "remote_chan_reserve_sat": "0",
                "initiator": "INITIATOR_REMOTE",
                "commitment_type": "STATIC_REMOTE_KEY"
            },
            "limbo_balance": "0",
            "commitments": {
                "local_txid": "",
                "remote_txid": "",
                "remote_pending_txid": "",
                "local_commit_fee_sat": "0",
                "remote_commit_fee_sat": "0",
                "remote_pending_commit_fee_sat": "0"
            }
        },
        {
            "channel": {
                "remote_node_pub": "030c3f19d742ca294a55c00376b3b355c3c90d61c6b6b39554dbc7ac19b141c14f",
                "channel_point": "36e2a2ec72da098f352fc44d2b54cb5bd8959a3f0d80d8248c4550d07afbcb7a:0",
                "capacity": "2000000",
                "local_balance": "0",
                "remote_balance": "0",
                "local_chan_reserve_sat": "0",
                "remote_chan_reserve_sat": "0",
                "initiator": "INITIATOR_LOCAL",
                "commitment_type": "STATIC_REMOTE_KEY"
            },
            "limbo_balance": "0",
            "commitments": {
                "local_txid": "",
                "remote_txid": "",
                "remote_pending_txid": "",
                "local_commit_fee_sat": "0",
                "remote_commit_fee_sat": "0",
                "remote_pending_commit_fee_sat": "0"
            }
        },
        {
            "channel": {
                "remote_node_pub": "036bddece3c19c9d2ed4d227ed759683e2348fdc48a2b4c682ce817f0928aa40c3",
                "channel_point": "9e8cd488071b2544a7b97184a9a9a0deef6d777111fc801203744ea4cc648dd5:0",
                "capacity": "100000",
                "local_balance": "0",
                "remote_balance": "0",
                "local_chan_reserve_sat": "0",
                "remote_chan_reserve_sat": "0",
                "initiator": "INITIATOR_REMOTE",
                "commitment_type": "STATIC_REMOTE_KEY"
            },
            "limbo_balance": "0",
            "commitments": {
                "local_txid": "",
                "remote_txid": "",
                "remote_pending_txid": "",
                "local_commit_fee_sat": "0",
                "remote_commit_fee_sat": "0",
                "remote_pending_commit_fee_sat": "0"
            }
        }
    ]
}
```

**Comment by guggero** [comment]
@otaviobps your problem doesn't seem to be related to this issue. I quickly checked all three channels and they all look to have been closed cooperatively a long time ago. Not sure why your node doesn't pick them up. Are you using Neutrino as a backend? Or a pruned bitcoind node? You might need to do a wallet rescan https://docs.lightning.engineering/lightning-network-tools/lnd/recovery#forced-in-place-rescan.

**Comment by otaviobps** [comment]
@guggero My LND is connected to my bitcoind fullnode, and the issue appeared after recovering from backup. For some reason, magically, LND removed the channels from the `waiting_close_channels`, not sure why, but now everything appears to be okay.

**Comment by guggero** [comment]
Ah, when you're recovering from backup then lnd is scanning the chain for each channel which can take a good while depending on the speed of your harddisk or SSD.

**Comment by JustAGuy12345** [comment]
Any updates on this case so far? Any potential paths for me to attempt?

**Comment by guggero** [comment]
There is nothing you can do on your end other than asking your peer to force close. The issue is investigating on how to prevent this from happening in the future. I didn't get to that yet, so no update.

**Comment by JustAGuy12345** [comment]
They have stopped responding to me, can you please urge them to do this. It will be on my mind forever

**Comment by JustAGuy12345** [comment]
https://1ml.com/node/0318ac9faa9629e7da08819bc8fe0dd2ae3044d69b1b2283a63479acffeb968483
they have an email listed but i think they blocked me

**Comment by guggero** [comment]
I've reached out to them and the channel is now being force-closed from their node. Your node should now be able to pull the funds (as soon as the force-close TX confirms).

**Comment by nemanja-z** [comment]
The channel is in the process of force closing, at least that's a message I got after trying to force close it.  It should be fixed soon. @JustAGuy12345

**Comment by tukac** [comment]
Hi guys, I have possibly similar problem. I have five channels in waiting_close_channels for many months now. My node is more than two years old. I have done blockchain rescan several times, no progress. I use full bitcoind. I can connect to those peers.

lncli pendingchannels
{
    "total_limbo_balance": "11092698",
    "pending_open_channels": [
    ],
    "pending_closing_channels": [
    ],
    "pending_force_closing_channels": [
    ],
    "waiting_close_channels": [
        {
            "channel": {
                "remote_node_pub": "022c260f9ad58196af280c80a96ec9eabf6404df59ff1a7553b0f381c875a29ba0",
                "channel_point": "4b1454408682e40eb7a6d75d01f7961014d352e0239d39b3c3c5cb86a304ba2f:1",
                "capacity": "11091534",
                "local_balance": "0",
                "remote_balance": "11087883",
                "local_chan_reserve_sat": "110915",
                "remote_chan_reserve_sat": "110915",
                "initiator": "INITIATOR_REMOTE",
                "commitment_type": "LEGACY"
            },
            "limbo_balance": "0",
            "commitments": {
                "local_txid": "6e987aac5c5a5cb7d6b734f6b855e053ffd3aac829f8783ebc460e9e78141bb9",
                "remote_txid": "1f05c29e5886e008975ea30943e958605ee6017c0a3ae3291f02d4c64f847991",
                "remote_pending_txid": "",
                "local_commit_fee_sat": "3651",
                "remote_commit_fee_sat": "3651",
                "remote_pending_commit_fee_sat": "0"
            }
        },
        {
            "channel": {
                "remote_node_pub": "02459b759a62bc3ebfe98a320da237944cc4f35456384bd8fdefa7d0340c75f46f",
                "channel_point": "04272ad3da5e8d070e5cd2defd598c262edce39e32246f11cb6a18687429d012:1",
                "capacity": "2472817",
                "local_balance": "0",
                "remote_balance": "2469166",
                "local_chan_reserve_sat": "24728",
                "remote_chan_reserve_sat": "24728",
                "initiator": "INITIATOR_REMOTE",
                "commitment_type": "LEGACY"
            },
            "limbo_balance": "0",
            "commitments": {
                "local_txid": "e68b07a3016ddfda07015e27fc7ce84195df8334e2370d0044f9c933da587ae2",
                "remote_txid": "099375df83edbf1ce7756a38973ec93fa52f9edcc84695f79200a955380df3ae",
                "remote_pending_txid": "",
                "local_commit_fee_sat": "3651",
                "remote_commit_fee_sat": "3651",
                "remote_pending_commit_fee_sat": "0"
            }
        },
        {
            "channel": {
                "remote_node_pub": "0279c22ed7a068d10dc1a38ae66d2d6461e269226c60258c021b1ddcdfe4b00bc4",
                "channel_point": "511f2273d1956cde67f17d83e7591c885bbf2e7fc82669f1ab54badeee3535b2:0",
                "capacity": "4995940",
                "local_balance": "0",
                "remote_balance": "4992289",
                "local_chan_reserve_sat": "49959",
                "remote_chan_reserve_sat": "49959",
                "initiator": "INITIATOR_REMOTE",
                "commitment_type": "LEGACY"
            },
            "limbo_balance": "0",
            "commitments": {
                "local_txid": "a1e5b1fe216a38a0326a78e8f0f5039c3220170a4404d93b3a9f06d612777914",
                "remote_txid": "9f011d0e8b059605299539698de5637be6cb4a233a7a11f66345f81ed790e489",
                "remote_pending_txid": "",
                "local_commit_fee_sat": "3651",
                "remote_commit_fee_sat": "3651",
                "remote_pending_commit_fee_sat": "0"
            }
        },
        {
            "channel": {
                "remote_node_pub": "030995c0c0217d763c2274aa6ed69a0bb85fa2f7d118f93631550f3b6219a577f5",
                "channel_point": "72150d90cfb6b716c7c47918176f8645a3a87a4fbff3451941fd7b3a793e3e42:1",
                "capacity": "10000000",
                "local_balance": "9996349",
                "remote_balance": "0",
                "local_chan_reserve_sat": "100000",
                "remote_chan_reserve_sat": "100000",
                "initiator": "INITIATOR_LOCAL",
                "commitment_type": "LEGACY"
            },
            "limbo_balance": "9996349",
            "commitments": {
                "local_txid": "441cdf948e0303582a72df15d43184faf0c327149ccbb3bc3b24e1a717916b3b",
                "remote_txid": "e3213e6727b17519a4532e7adc5a1bffbd64314c0638b662772da3a23f743191",
                "remote_pending_txid": "",
                "local_commit_fee_sat": "3651",
                "remote_commit_fee_sat": "3651",
                "remote_pending_commit_fee_sat": "0"
            }
        },
        {
            "channel": {
                "remote_node_pub": "03c436af41160a355fc1ed230a64f6a64bcbd2ae50f12171d1318f9782602be601",
                "channel_point": "3cdf2d7ed75ecd3869c230ffbd3aa7800c31f0e69dc8dc3cb3bd4d4105bc1838:0",
                "capacity": "1100000",
                "local_balance": "1096349",
                "remote_balance": "0",
                "local_chan_reserve_sat": "11000",
                "remote_chan_reserve_sat": "11000",
                "initiator": "INITIATOR_LOCAL",
                "commitment_type": "LEGACY"
            },
            "limbo_balance": "1096349",
            "commitments": {
                "local_txid": "3909ece1f106111f593e306824c8af65c7c690d6e55083d71e8cba87f7252922",
                "remote_txid": "abbcf7918cb778ddcfb7879fe56d178cf6604cd330afe1447d45af319a8f0e17",
                "remote_pending_txid": "",
                "local_commit_fee_sat": "3651",
                "remote_commit_fee_sat": "3651",
                "remote_pending_commit_fee_sat": "0"
            }
        }
    ]
}

Why are those channels still waiting close?

**Comment by guggero** [comment]
@tukac did you recover those channels using the SCB/channel.backup file? Are you using a pruned bitcoind node? Looks like the channels all closed a long time ago but just don't update in `lnd`.

**Comment by tukac** [comment]
No SCB usage. I use full bitcoind 0.21.1.

**Comment by guggero** [comment]
Can you restart your node and grep the log for one of the channel points please? Not sure why they aren't being resolved.

**Comment by guggero** [comment]
Interesting, thanks for the log, @tukac. I just noticed that all these channels are `LEGACY` channels, so `lnd` will need some information from the peer (the `commitment point`) to derive the force-close addresses. But the peers apparently already removed the information for those channels as they were closed a long time ago.

I assume all funds are accounted for in the wallet, correct? You just want to get rid of the channels being displayed?
Can you try if you can use `lncli abandonchannel` on those channels? If that works, that's probably the easiest way to get rid of them since the funds should all be safe anyway.

**Comment by tukac** [comment]
Thanks for the reply. Yes, all funds are accounted for. Can you please suggest how to correctly formulate `lncli abandonchannel` for channel_point: 4b1454408682e40eb7a6d75d01f7961014d352e0239d39b3c3c5cb86a304ba2f:1 ?

**Comment by guggero** [comment]
`lncli abandonchannel --funding_txid 4b1454408682e40eb7a6d75d01f7961014d352e0239d39b3c3c5cb86a304ba2f --output_index 1 --i_know_what_i_am_doing`

**Comment by tukac** [comment]
Thanks, it did job for 3 of 5 channels. For remaining two I get:

```
lncli abandonchannel --funding_txid 511f2273d1956cde67f17d83e7591c885bbf2e7fc82669f1ab54badeee3535b2 --output_index 1 --i_know_what_i_am_doing
[lncli] rpc error: code = Unknown desc = unable to find closed channel summary
```

```
lncli abandonchannel --funding_txid 3cdf2d7ed75ecd3869c230ffbd3aa7800c31f0e69dc8dc3cb3bd4d4105bc1838 --output_index 1 --i_know_what_i_am_doing
[lncli] rpc error: code = Unknown desc = unable to find closed channel summary
```

**Comment by tukac** [comment]
I have another issue with this old node (from 07/2019). Every restart it is doing some task but I don't know what it is. It has been doing this task as long as I remember. This task feels like blockchain rescan. On RPi 4b it lasts for about two days (uses only one core) and around 60Mb/s SSD read and write constantly all the time.

Can you please advise what to grep for?

**Comment by guggero** [comment]
> For remaining two I get:

The number after the `:` in the channel point corresponds to the `--output_index`. So you need to update that in your commands.

> Every restart it is doing some task but I don't know what it is.

My guess is that `lnd` was scanning the chain for those 5 channels, which on a RPi 4 can take quite a while. I assume the problem will go away, now that you've removed those channels. If not, please open another issue with logs attached running on `debuglevel=debug`.
